src/tests/run.proj

```diff
@@ -235,10 +239,78 @@ namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").
                 runningInWindows = (operatingSystem != null && operatingSystem.StartsWith("Windows"))%3B
             }
         }
-
 ]]>
       </_XunitProlog>
+      <_XunitProlog Condition=" '$(_XunitProlog)'=='' and '$(IsMobile)'=='true' ">
+        <![CDATA[
+using Xunit%3B
+using Xunit.Abstractions%3B
+using System%3B
+using System.Collections.Generic%3B
+using System.Diagnostics%3B
+using System.Reflection%3B
+using System.Text.RegularExpressions%3B
+using CoreclrTestLib%3B
+using System.IO%3B
+
+namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").Replace("-","_"))
+{
+        public class _Global : IDisposable
+        {
+            public bool runningInWindows%3B
+            public string reportBase%3B
+            public string testBinaryBase%3B
+            public string coreRoot%3B
+            public string category%3B
+            public string helixUploadRoot%3B
+            public MobileAppHandler handler = new MobileAppHandler()%3B
+
+            public _Global()
+            {
+                reportBase = System.Environment.GetEnvironmentVariable(%22XunitTestReportDirBase%22)%3B
+                testBinaryBase = System.IO.Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)%3B
+                coreRoot = System.Environment.GetEnvironmentVariable(%22CORE_ROOT%22)%3B
+                category = "$([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").Replace("-","_"))"%3B
+                helixUploadRoot = System.Environment.GetEnvironmentVariable(%22HELIX_WORKITEM_UPLOAD_ROOT%22)%3B
+                if (!String.IsNullOrEmpty(helixUploadRoot)) {
+                    reportBase = System.IO.Path.Combine(System.IO.Path.GetFullPath(helixUploadRoot), "Reports")%3B
+                }
+
+                if (String.IsNullOrEmpty(reportBase)) {
+                    reportBase = System.IO.Path.Combine(testBinaryBase, %22Reports%22)%3B
+                }
+                else
+                {
+                    reportBase = System.IO.Path.GetFullPath(reportBase)%3B
+                }
+
+                if (String.IsNullOrEmpty(coreRoot)) {
+                    throw new ArgumentException("Environment variable CORE_ROOT is not set")%3B
+                }
+
+                coreRoot = System.IO.Path.GetFullPath(coreRoot)%3B
```

Comment:
```suggestion
                reportBase = Environment.GetEnvironmentVariable(%22XunitTestReportDirBase%22)%3B
                testBinaryBase = GetDirectoryName(Assembly.GetExecutingAssembly().Location)%3B
                coreRoot = Environment.GetEnvironmentVariable(%22CORE_ROOT%22)%3B
                category = "$([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").Replace("-","_"))"%3B
                helixUploadRoot = Environment.GetEnvironmentVariable(%22HELIX_WORKITEM_UPLOAD_ROOT%22)%3B
                if (!String.IsNullOrEmpty(helixUploadRoot)) {
                    reportBase = Combine(GetFullPath(helixUploadRoot), "Reports")%3B
                }

                if (String.IsNullOrEmpty(reportBase)) {
                    reportBase = Combine(testBinaryBase, %22Reports%22)%3B
                }
                else
                {
                    reportBase = GetFullPath(reportBase)%3B
                }

                if (String.IsNullOrEmpty(coreRoot)) {
                    throw new ArgumentException("Environment variable CORE_ROOT is not set")%3B
                }

                coreRoot = GetFullPath(coreRoot)%3B
```

Reply:
@fanyang-mono oops, I messsed up this comment. Shouldn't remove `Path.`

Reply:
no problem, I will fix it.